### PR TITLE
[FIX] various: use Markup for empty list help

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Some cases require usage of Markup in order to correctly display HTML content
and avoid escaped or invalid statements.

Task-3074357

### CODE REVIEW

File: addons/hr_recruitment/models/hr_recruitment.py [hunk 1]
```diff
@@ -440,22 +441,20 @@ def get_empty_list_help(self, help_message):
         else:
             alias_id = False
 
-        nocontent_values = {
+        nocontent_body = Markup(
+            """
+<p class="o_view_nocontent_empty_folder">%(help_title)s</p>
+<p>%(para_1)s<br/>%(para_2)s</p>""" % {
```
Comment (mart-e) [inline addons/hr_recruitment/models/hr_recruitment.py:447]:
```suggestion
<p>%(para_1)s<br/>%(para_2)s</p>""") % {
```
you can move the labels outside of markup if it does not contain html code

**Review by mart-e — COMMENTED** [review]
@robodoo override=ci/security
Reason: adding markup for moar escaping, escaping document name in help

File: addons/hr_recruitment/models/hr_recruitment.py (hunk 1 above)
Comment (tde-banana-odoo) [reply addons/hr_recruitment/models/hr_recruitment.py]: Done Sir.

File: addons/crm/models/crm_lead.py [hunk 2]
```diff
@@ -1254,10 +1255,12 @@ def get_empty_list_help(self, help_message):
...
         ], limit=1)
+
         if alias_record and alias_record.alias_domain and alias_record.alias_name:
-            email = '%s@%s' % (alias_record.alias_name, alias_record.alias_domain)
-            email_link = "<b><a href='mailto:%s'>%s</a></b>" % (email, email)
-            sub_title = _('Use the top left <i>Create</i> button, or send an email to %s to test the email gateway.') % (email_link)
+            email = f'{alias_record.alias_name}@{alias_record.alias_domain}'
+            sub_title = Markup(_('Use the top left <i>Create</i> button, or send an email to %(email_link)s to test the email gateway.')) % {
+                'email_link': Markup("<b><a href='mailto:%s'>%s</a></b>") % (email, email),
+            }
```
Comment (std-odoo) [inline addons/crm/models/crm_lead.py:1263]: We can call Markuponly once
```suggestion
            sub_title = Markup(_('Use the top left <i>Create</i> button, or send an email to <b><a href='mailto:%s'>%s</a></b> to test the email gateway.')) % (email, email)
```

File: addons/mail/models/mail_gateway_allowed.py [hunk 3]
```diff
@@ -35,7 +36,7 @@ def get_empty_list_help(self, help_message):
         LOOP_MINUTES = int(get_param('mail.gateway.loop.minutes', 120))
         LOOP_THRESHOLD = int(get_param('mail.gateway.loop.threshold', 20))
 
-        return _('''
+        return Markup(_('''
```
Comment (std-odoo) [inline addons/mail/models/mail_gateway_allowed.py:39]: Markup is useless here (we still pass the argument to `_` not to `Markup`, and they are `int` so it's ok)

or if we really want to return a Markup object, we should give the argument to Markup itself (not `_`)

**Review by std-odoo — COMMENTED** [review]
@tde-banana-odoo Hi :) 

I just have 2 small comments, otherwise LGTM :)

File: addons/crm/models/crm_lead.py (hunk 2 above)
Comment (tde-banana-odoo) [reply addons/crm/models/crm_lead.py:1263]: Purpose is to get the href out of the string to translate actually :) that way we lessen potential issues if people wrongly translate html. Few HTML is ok but when possible ...

File: addons/mail/models/mail_gateway_allowed.py (hunk 3 above)
Comment (tde-banana-odoo) [reply addons/mail/models/mail_gateway_allowed.py:39]: Done, just to encore usage of markup I kept the Markup and gave parameters to him.

File: addons/crm/models/crm_lead.py (hunk 2 above)
Comment (std-odoo) [reply addons/crm/models/crm_lead.py:1263]: ah ok :) ok for me so :)

**Review by std-odoo — APPROVED** [review]

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**